packages/table/src/regions.ts

```diff
@@ -211,16 +221,37 @@ export class Regions {
 
     /**
      * Replaces the region at the end of a cloned copy of the supplied region
-     * array.
+     * array, or at the specific index if one is provided.
      */
-    public static update(regions: IRegion[], region: IRegion) {
+    public static update(regions: IRegion[], region: IRegion, index?: number) {
```

Comment:
`index` param necessary now that we need to modify regions at particular indices in the `selectedRegions` array.